src/bc_protocol/media_packet.rs

```diff
@@ -0,0 +1,381 @@
+            MediaDataKind::VideoDataPframe => MediaData::bytes_to_size(&data[8..12]),
+            MediaDataKind::AudioDataAac => MediaData::bytes_to_size(&data[4..6]),
+            MediaDataKind::AudioDataAdpcm => MediaData::bytes_to_size(&data[4..6]),
+            MediaDataKind::InfoData => 0, // The bytes in MediaData::bytes_to_size(&data[4..8]) seem to be the size of the header
+            MediaDataKind::Unknown | MediaDataKind::Invalid | MediaDataKind::Continue => data.len(),
+        }
+    }
+
+    pub fn data_size(&self) -> usize {
+        MediaData::data_size_from_raw(&self.data)
+    }
+
+    pub fn pad_size_from_raw(data: &[u8]) -> usize {
+        let data_size = MediaData::data_size_from_raw(data);
+        match data_size % 8 {
+            0 => 0,
+            n => 8 - n,
+        }
+    }
+
+    pub fn pad_size(&self) -> usize {
+        MediaData::data_size_from_raw(&self.data)
+    }
+
+    fn bytes_to_size(bytes: &[u8]) -> usize {
+        match bytes.len() {
+            // 8 Won't fit into usize on a 32-bit machine
+            4 => u32::from_le_bytes(bytes.try_into().expect("slice with incorrect length"))
+                .try_into()
+                .expect("u32 won't fit into usize"),
+            2 => u16::from_le_bytes(bytes.try_into().expect("slice with incorrect length"))
+                .try_into()
+                .expect("u16 won't fit into usize"),
+            1 => u8::from_le_bytes(bytes.try_into().expect("slice with incorrect length"))
+                .try_into()
+                .expect("u8 won't fit into usize"),
+            _ => unreachable!(),
+        }
+    }
+
+    pub fn kind_from_raw(data: &[u8]) -> MediaDataKind {
+        // When calling this ensure you have enough data for header_size +2
+        // Else full_header_check_from_kind will fail because we check the
+        // First two bytes after the header for the audio stream
+        // Since AAC and ADMPC streams start in a predicatble manner
+        if data.len() < 4 {
+            return MediaDataKind::Invalid;
+        }
+        const MAGIC_VIDEO_INFO_V1: &[u8] = &[0x31, 0x30, 0x30, 0x31];
+        const MAGIC_VIDEO_INFO_V2: &[u8] = &[0x31, 0x30, 0x30, 0x32];
+        const MAGIC_AAC: &[u8] = &[0x30, 0x35, 0x77, 0x62];
+        const MAGIC_ADPCM: &[u8] = &[0x30, 0x31, 0x77, 0x62];
+        const MAGIC_IFRAME: &[u8] = &[0x30, 0x30, 0x64, 0x63];
+        const MAGIC_PFRAME: &[u8] = &[0x30, 0x31, 0x64, 0x63];
+
+        let magic = &data[..4];
+        let kind = match magic {
+            MAGIC_VIDEO_INFO_V1 | MAGIC_VIDEO_INFO_V2 => MediaDataKind::InfoData,
+            MAGIC_AAC => MediaDataKind::AudioDataAac,
+            MAGIC_ADPCM => MediaDataKind::AudioDataAdpcm,
+            MAGIC_IFRAME => MediaDataKind::VideoDataIframe,
+            MAGIC_PFRAME => MediaDataKind::VideoDataPframe,
+            _ if data.len() == CHUNK_SIZE => MediaDataKind::Continue,
```

Comment:
This is the only thing I don't like about this current implementation. Should not assume that the sender is sending `CHUNK_SIZE` packets as a continuation - logically the media packet payloads simply form a byte stream and we should be able to read any incoming chunk size. (If this sounds a lot like `Read`, it's because it is.)

The better strategy is to keep adding whatever data is received to the outstanding packet.

That said, the API of this module is good - "get next media packet" makes it easy to change this after this PR is merged. So if you want me to look at it down the road, I'm okay with that, since it works with existing camera firmware.

Reply:
The actual implementation dosent depend on finding a continuation packet. It works like you say by pulling more data and adcanicng. so we could just drop this type.

Reply:
I've completely removed the continuation type as its all unneed and hidden by the get next media packet anyway